### ISSUE TITLE
2 Bugs: no tag for "quartier" even tho "quartier" is there; cannot access an existing place

### ISSUE THREAD

**Issue by MaudeFJ** [culturecreates/tech-support-tickets#100]
Team: 2 issues
Related issue - already logged in #99 
for this bug, Maude had to create a duplicate place because she cannot access the original place.
I can search an see the two places: 
https://cms.footlight.io/dashboard/63457978637093005071a053/places?page=1&order=asc&sortBy=name&query=salle+anais 
I tried and cannot access the old one either (which uses the character ä and is linked to Artsdata):
https://cms.footlight.io/dashboard/63457978637093005071a053/places/671fc4e6017105006374409d
**you can test and resolve this in the  other issue (#99)**

### MAIN issue for this bug
Sometimes there is no tag for "quartier" in the Place component in Events even though a "quartier" is already selected in the Place entity there. This issue happened with the new Salle Anais Allard Rousseau Place she created (mentioned in first issue).

**Steps:**
In the event, the place card is displayed but the neighborhood (quartier) tag is not included (displayed) in the component.
The user clicks Edit.  In the Place form, the quartier is already selected.
User returns tothe event and the quartier tag is now displayed.

**Expected behavior:** the selected neighborhood/quartier should be displayed in the Place component in Events, like in this example (Trois Rivieres is the quartier):
<img width="441" alt="Screenshot 2024-11-05 at 11 30 37 AM" src="https://github.com/user-attachments/assets/ca90f0b3-f6fb-4ae8-a47a-fa58821dca64">

_Note from Maude
I need to go back and forth in the place creation page for it to appear in the event... 
(I don't have screen cap cause it's there now but I had this issue with the new Salle Anais Allard Rousseau I created (cause I could not access the old one)_

**Comment by troughc** [comment]
@MaudeFJ can you please add detailed steps - I don't understand the quartier bug.

**Comment by MaudeFJ** [comment]
Dans l'événement je vois la petite fiche du lieu mais il n'y a pas de tag quartier, je clic sur éditer, puis dans la grande fiche le quartier est déjà là, je retourne sur l'événement puis le tag quartier est là.
J'aimerais que le tag quartier soit déjà là s'il est déjà là.



Maude Fraser

Responsable de Signé Laval  ​

450 639-3550 – poste 7
387 rue Laurier, Laval, QC H7N 2P3
  [cid:0d3da6bc-d779-4cde-b907-b2f991ab5537]

Suivez-nous sur Facebook<https://www.facebook.com/signelaval/>, Twitter<https://twitter.com/Signe_Laval>, Instagram<https://www.instagram.com/> et Youtube<https://www.youtube.com/channel/UCR8T8HTc83UHH7CPhxTlxXw?view_as=subscriber>.

________________________________
De : troughc ***@***.***>
Envoyé : 4 novembre 2024 16:09
À : culturecreates/tech-support-tickets ***@***.***>
Cc : Maude Fraser ***@***.***>; Mention ***@***.***>
Objet : Re: [culturecreates/tech-support-tickets] Bug no tag for "quartier" even tho "quartier" is there (Issue #100)


@MaudeFJ<https://github.com/MaudeFJ> can you please add detailed steps - I don't understand the quartier bug.

**Comment by MaudeFJ** [comment]
<img width="500" alt="Capture d’écran, le 2024-11-07 à 10 28 55" src="https://github.com/user-attachments/assets/28edbf39-53b1-4ea6-b411-f6e79351c8e8">
<img width="683" alt="Capture d’écran, le 2024-11-07 à 10 29 04" src="https://github.com/user-attachments/assets/c94615c5-68d9-4483-825e-cc6e74ee6ad2">
Here is a picture exemple !

**Comment by AbhishekPAnil** [comment]
@troughc Were you able to reproduce the bug,if so can you guide me through the steps.Thanks.

**Comment by troughc** [comment]
@MaudeFJ do you have any other events that show location cards (with a place)s that are NOT displaying the 'Quartier' tag in the card? If you do, can you please put the link here and please do NOT fix it. 
Also, do you have the steps you did when this happened?

**Comment by MaudeFJ** [comment]
I don't know,
 i can't know for sure if they do in fact have the tag, before I go on edit mode, and then when I go back in the event the tag appears... 
 Right now there is one in "attente d'approbation", nothing is in the little place box in the event. Is it because the person did not put all the required info, or is it a bug... I don't know

**Comment by troughc** [comment]
@AbhishekPAnil. the event that @MaudeFJ is referring to this one: https://cms.footlight.io/dashboard/63457978637093005071a053/events/add-event/672eb7cdcd518800637922d5
This event is linked to this place:
https://cms.footlight.io/dashboard/63457978637093005071a053/places/672eb7cdcd518800637922cf

This place is missing the 'quartier' filled, even though it is mandatory. The place was created by a guest user (probably by quick create menu)

I just quick created a place using the Quick Create menu as a test, the place is named [Place test](https://cms.footlight.io/dashboard/63457978637093005071a053/places/6732181f2bb3dc0063140c2a)

I was able to Quick create it without adding a district / quartier.  Because the quartier is NOT mandatory in the quick create menu (but it does appear). I believe it should be a mandatory field

Steps to reproduce:
in an event form, open the quick create menu and create a new place. Do not add the quartier/district and then click save. The place is created (no warning to add a quartier).
Expected behavior: asterisk beside the district/quartier in quick create; and it should be mandatory in quick create.

**Comment by MaudeFJ** [comment]
Est-ce que je peux faire les modifications et publier l'événement ?

**Comment by AbhishekPAnil** [comment]
> [@AbhishekPAnil](https://github.com/AbhishekPAnil). the event that [@MaudeFJ](https://github.com/MaudeFJ) is referring to this one: https://cms.footlight.io/dashboard/63457978637093005071a053/events/add-event/672eb7cdcd518800637922d5 This event is linked to this place: https://cms.footlight.io/dashboard/63457978637093005071a053/places/672eb7cdcd518800637922cf
> 
> This place is missing the 'quartier' filled, even though it is mandatory. The place was created by a guest user (probably by quick create menu)
> 
> I just quick created a place using the Quick Create menu as a test, the place is named [Place test](https://cms.footlight.io/dashboard/63457978637093005071a053/places/6732181f2bb3dc0063140c2a)
> 
> I was able to Quick create it without adding a district / quartier. Because the quartier is NOT mandatory in the quick create menu (but it does appear). I believe it should be a mandatory field
> 
> Steps to reproduce: in an event form, open the quick create menu and create a new place. Do not add the quartier/district and then click save. The place is created (no warning to add a quartier). Expected behavior: asterisk beside the district/quartier in quick create; and it should be mandatory in quick create.

@troughc 
If thats the case ,then I think we may need to handle the cases of mandatory fields in Quick creates (for organization,person as well ).If a field is added as mandatory by the user ,then the field should be automatically be added to the quick create form as well.Do you agree ?

**Comment by troughc** [comment]
@AbhishekPAnil Yes. I believe that the field is already added to Quick Create, it just doesn't look like it was made to be mandatory. Is that correct?

**Comment by troughc** [comment]
> Est-ce que je peux faire les modifications et publier l'événement ?

@MaudeFJ the team is going to finish the work tomorrow, but I think it's okay if you review and publish the event from your guest user. So yes, please feel free to do so.

FYI: the reason the district isn't showing up is because the user didn't add it in the Quick Create modal. Our fix will make sure that 'Quartier' is mandatory in the quick create modal so this doesn't happen again. 
However, there is one exception: places that are added by an AGGREGATOR (new places that are added with an Event as its location) are allowed to be added without the quartier field filled. This is so that your events won't be blocked if that data is missing from the location. In this case, you will have to add the district yourself. You will be able to see it when you review the event (it will be missing from the card).

**Comment by troughc** [comment]
@AbhishekPAnil I am adding you back as an assignee to follow up on our discussion.

**Comment by AbhishekPAnil** [comment]
@troughc Please test in this [PR](https://pr-1444.dla06mbe0wyl7.amplifyapp.com/) if "regions" are shown as required only if its added as mandatory.

**Comment by AbhishekPAnil** [comment]
The validation check should also be done in the  backend for the same.Now a user is able to quick create a place in a Signe laval calendar without regions where "region" field is mandatory.Please assign this back to the backend team once the frontend review is complete.

**Comment by troughc** [comment]
Tested the front end for 'region'. This is working well.
Assigning to the backend team. 
cc @AbhishekPAnil

**Comment by suhail-aliyar** [comment]
The backend change will be addressed in #1445

**Comment by MaudeFJ** [comment]
It happend again !! 
<img width="807" alt="Capture d’écran, le 2024-11-22 à 09 56 37" src="https://github.com/user-attachments/assets/f9bb43b5-4e8f-428c-ac1f-3583926ef9c5">
<img width="925" alt="Capture d’écran, le 2024-11-22 à 09 56 50" src="https://github.com/user-attachments/assets/d8dbc52b-3b30-41cc-9e4a-53006d5a1e4f">
I just quit the page with the x so it would stay like that https://cms.footlight.io/dashboard/63457978637093005071a053/events/673f6b93b67edf00649110c5
 but the event is starting today so I need to publish it ASAP

**Comment by troughc** [comment]
@sahalali FYI - Maude's problem with regions happened again. Is this because we still have to push #1445 to production for this to be fixed?

**Comment by suhail-aliyar** [comment]
Please review this.

**Comment by dev-aravind** [comment]
@sahalali @troughc Looks good to me, but I found a [place](https://cms.footlight.io/dashboard/63457978637093005071a053/places/66843b579d0a290064b86d2b) with no postal address and it is not linked to artsdata and any of the footlight entities. 

Update: I found the [artsdata entity](https://kg.artsdata.ca/entity.html?uri=http%3A%2F%2Fkg.artsdata.ca%2Fresource%2FK16-325) for this place and will be fixing this entity ASAP.

Update 2: The team had a discussion on this and decided to delete the place instead.

**Comment by dev-aravind** [comment]
@troughc The place is deleted right now and the bug is now fixed in production.

**Comment by troughc** [comment]
@MaudeFJ the reason you had the issue again is because another issue had to be pushed to production (which has now happened). So the issue is fixed.

**Comment by troughc** [comment]
@MaudeFJ this is now fixed so I am closing the issue.
cc @sahalali

**Comment by MaudeFJ** [comment]
People are still able to create place without the mandatory field "quartier"

**Comment by troughc** [comment]
@MaudeFJ I was not able to. I tried in Places and failed and I also tried in quick create and I could not proceed - see screenshot. Could you please provide the steps to reproduce this bug?
thanks!

<img width="905" alt="Screenshot 2025-01-07 at 9 15 43 PM" src="https://github.com/user-attachments/assets/c934b098-03a0-42e2-9829-bb1650af7214" />

**Comment by MaudeFJ** [comment]
Hum its when i reviewed an event i've seen a place without a quartier, but maybe it was created before and still carried the absence of quartier..
